### PR TITLE
Delete config statuses prior refreshing

### PR DESCRIPTION
**Pull request by lzap**
Okay it looks like there are still old host status objects for legacy configuration in your database and the recalculation of statuses does not remove them.

https://community.theforeman.org/t/migrating-reports-to-the-new-format-in-foreman-3-2/25846/9

### CODE REVIEW

File: lib/tasks/migrate.rake [hunk 1]
```diff
@@ -220,11 +220,19 @@ namespace :host_reports do
   This task refreshes all host statuses and global statuses.
   END_DESC
   task :refresh => :environment do
+    # ensure the class is loaded before refresh starts to avoid hash modifycation during loop
+    HostStatus::HostReportStatus
```
Comment (lzap) [inline lib/tasks/migrate.rake]: I think this is causing "can't add a new key into hash during iteration"

File: lib/tasks/migrate.rake [hunk 2]
```diff
@@ -220,11 +220,19 @@ namespace :host_reports do
...
+
     Rails.logger.level = Logger::ERROR
     Foreman::Logging.logger('permissions').level = Logger::ERROR
     Foreman::Logging.logger('audit').level = Logger::ERROR
     User.without_auditing do
       User.as_anonymous_admin do
+        # delete old and new statuses
+        HostStatus::Status.where(type: "HostStatus::ConfigurationStatus").delete_all
+        HostStatus::Status.where(type: "HostStatus::HostReportStatus").delete_all
```
Comment (lzap) [inline lib/tasks/migrate.rake:233]: This needs to be done prior refresh as refresh does not remove any statuses it only recalculates values or adds new ones.

**Comment by lzap** [comment]
Confirmed to be working by the user. @avitova you want to review?

File: lib/tasks/migrate.rake (hunk 1 above)
Comment (avitova) [reply lib/tasks/migrate.rake]: Tiny nitpick: modification?

**Comment by avitova** [comment]
Tested, works as I would expected.

**Comment by lzap** [comment]
Thanks a bunch! Merging.

**Pull request merged**